packages/plugin-hardhat/src/index.ts

```diff
@@ -157,6 +132,71 @@ if (tryRequire('@nomiclabs/hardhat-etherscan')) {
+  const { makeUpgradeProxy } = require('./upgrade-proxy');
+  const { makeValidateImplementation } = require('./validate-implementation');
+  const { makeValidateUpgrade } = require('./validate-upgrade');
+  const { makeDeployImplementation } = require('./deploy-implementation');
+  const { makeDeployContract } = require('./deploy-contract');
+  const { makePrepareUpgrade } = require('./prepare-upgrade');
+  const { makeDeployBeacon } = require('./deploy-beacon');
+  const { makeDeployBeaconProxy } = require('./deploy-beacon-proxy');
+  const { makeUpgradeBeacon } = require('./upgrade-beacon');
+  const { makeForceImport } = require('./force-import');
+  const { makeChangeProxyAdmin, makeTransferProxyAdminOwnership, makeGetInstanceFunction } = require('./admin');
+  const { makeProposeUpgrade } = require('./platform/propose-upgrade');
+  const {
+    makeVerifyDeploy,
+    makeVerifyDeployWithUploadedArtifact,
+    makeGetVerifyDeployBuildInfo,
+    makeGetVerifyDeployArtifact,
+    makeGetBytecodeDigest,
+  } = require('./platform/verify-deployment');
+
+  return {
+    silenceWarnings,
+    deployProxy: makeDeployProxy(hre, platform),
+    upgradeProxy: makeUpgradeProxy(hre, platform), // block on platform
+    validateImplementation: makeValidateImplementation(hre),
+    validateUpgrade: makeValidateUpgrade(hre),
+    deployImplementation: makeDeployImplementation(hre, platform),
+    deployContract: makeDeployContract(hre, platform),
+    prepareUpgrade: makePrepareUpgrade(hre, platform),
+    deployBeacon: makeDeployBeacon(hre, platform), // block on platform
+    deployBeaconProxy: makeDeployBeaconProxy(hre, platform),
+    upgradeBeacon: makeUpgradeBeacon(hre, platform), // block on platform
+    deployProxyAdmin: makeDeployProxyAdmin(hre, platform), // block on platform
+    forceImport: makeForceImport(hre),
+    admin: {
+      getInstance: makeGetInstanceFunction(hre),
+      changeProxyAdmin: makeChangeProxyAdmin(hre, platform), // block on platform
+      transferProxyAdminOwnership: makeTransferProxyAdminOwnership(hre, platform), // block on platform
+    },
+    erc1967: {
+      getAdminAddress: (proxyAddress: string) => getAdminAddress(hre.network.provider, proxyAddress),
+      getImplementationAddress: (proxyAddress: string) => getImplementationAddress(hre.network.provider, proxyAddress),
+      getBeaconAddress: (proxyAddress: string) => getBeaconAddress(hre.network.provider, proxyAddress),
+    },
+    beacon: {
+      getImplementationAddress: (beaconAddress: string) =>
+        getImplementationAddressFromBeacon(hre.network.provider, beaconAddress),
+    },
+    proposeUpgrade: makeProposeUpgrade(hre, platform),
+    verifyDeployment: makeVerifyDeploy(hre),
+    verifyDeploymentWithUploadedArtifact: makeVerifyDeployWithUploadedArtifact(hre),
+    getDeploymentArtifact: makeGetVerifyDeployArtifact(hre),
+    getDeploymentBuildInfo: makeGetVerifyDeployBuildInfo(hre),
+    getBytecodeDigest: makeGetBytecodeDigest(hre),
```

Comment:
These are cloned from the `hardhat-defender` package.  We could make these only available for `platform.<fn>` and not `upgrades.<fn>`

However, `deployContract` and `proposeUpgrade` should be considered separately since they take the `platform` option.

`upgrades.deployContract` without `platform` option is not allowed anyways and would fail.  While the syntax `upgrades.deployContract(...{platform: true})` would be consistent with what we allow for `deployProxy`, I no longer see any real benefit to allowing it for this case (other than having one less import/require in the user's code).

`upgrades.proposeUpgrade` without `platform` option could make sense.  This would mean deploy the new implementation using ethers.js, then submit the proposal on Platform (even though Platform is not part of the name).

---
To summarize:

Support:
```
platform.deployContract
platform.verifyDeployment
platform.verifyDeploymentWithUploadedArtifact
platform.getDeploymentArtifact
platform.getDeploymentBuildInfo
platform.getBytecodeDigest
platform.proposeUpgrade
upgrades.proposeUpgrade
```

Do not support:
```
upgrades.deployContract
upgrades.verifyDeployment
upgrades.verifyDeploymentWithUploadedArtifact
upgrades.getDeploymentArtifact
upgrades.getDeploymentBuildInfo
upgrades.getBytecodeDigest
```

Let me know if you agree.

Reply:
What do you think of putting `platform` under `upgrades`?
So we would have
```
upgrades: {
  deployProxy: ...,
  ...,
  platform: {
    deployProxy: ..., // aliased with { platform: true }
    deployContract: ..., // only at this level
    ...
  }
}
```

Reply:
Removed everything related to `verifyDeployment` since that is handled automatically by the Platform deploy API.

I don't think `upgrades.platform.deployContract` makes sense since it is unrelated to upgradeable contracts.

Now since only `deployContract` and `proposeUpgrade` are new, I changed it to this:
```
upgrades: {
  deployProxy: ...,
  prepareUpgrade: ...,
  ...,
},
platform: {
  deployProxy: ..., // aliased with { platform: true }
  prepareUpgrade: ..., // aliased with { platform: true }
  ...,
  deployContract: ..., // only at this level
}
```